src/Nethermind/Nethermind.State/StateProvider.cs

```diff
@@ -36,7 +39,8 @@ internal class StateProvider
 
         private readonly List<Change> _keptInCache = new();
         private readonly ILogger _logger;
-        private readonly IKeyValueStore _codeDb;
+        private readonly IKeyValueStoreWithBatching _codeDb;
+        private Dictionary<ValueHash256, byte[]> _codeBatch;
```

Comment:
are 32byte keys optimal?

Reply:
Not particularly but is how they are passed around so unless we allocate a Hash256 on each access 🤷‍♂️

Reply:
Added Alternate lookup